src/media/mod.rs

```diff
@@ -742,7 +751,7 @@ impl UploadError {
     }
 }
 
-/// Represents the status of an `UploadFuture` when it encountered an error.
+/// Represents the status of an `UploadFuture`.
```

Comment:
`UploadState` is only ever used as part of an `UploadError`, so it's only ever useful for the user in that context. I'm deliberately hiding the full `UploadInner`, which is the true status of the Future.

Reply:
The problem with this is that it actually contradicts to its name and description of Complete.

>The `UploadFuture` was fully completed, or previously encountered an error that dropped it out of the upload process.

While you're seeing in UploadError only,  it is still representation of UploadFuture status, not the error status.

Reply:
That's a good point. You can keep this change in.